### PR TITLE
Add source code access for showcases

### PR DESCRIPTION
**Pull request by waliid**
# Description

This PR enables easy access to the source code for showcases.

# Changes made

- Added **GitHub** scheme to the `LSApplicationQueriesSchemes` in the **Info.plist** file.
- Added `SourceCodeButton` for a **GitHub** button with a callback.
- Defined a protocol `SourceCodeViewable` with a static property `filePath`.
- Added `sourceCode` modifier for easy access to source code.

# Demo

https://github.com/SRGSSR/pillarbox-apple/assets/3347810/c6d5be97-0a81-48cd-9e04-03af7f72260c

# Checklist

- [x] APIs have been properly documented (if relevant).
- [x] The documentation has been updated (if relevant).
- [x] New unit tests have been written (if relevant).
- [x] The demo has been updated (if relevant).
- [x] The playground has been updated (if relevant).

### CODE REVIEW

**Review by MGaetan89 — APPROVED** [review]

File: Demo/Sources/Showcase/SourceCodeViewable.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        Button {
+            perform()
+        } label: {
+            HStack {
+                Text("GitHub")
+            }
+        }
+        .tint(.secondary)
+    }
+}
```
Comment (defagos) [inline Demo/Sources/Showcase/SourceCodeViewable.swift:22]:
```suggestion
private struct SourceCodeButton: View {
    private let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack {
                Text("GitHub")
            }
        }
        .tint(.secondary)
    }

    init(perform action: @escaping () -> Void) {
        self.action = action
    }
}
```

File: Demo/Sources/Showcase/SourceCodeViewable.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+protocol SourceCodeViewable {
+    static var filePath: String { get }
+}
+
+extension View {
+    @ViewBuilder
+    func sourceCode(of objectType: (any SourceCodeViewable.Type)?) -> some View {
+#if os(iOS)
+        if let objectType, let url = gitHubUrl(for: objectType) {
```
Comment (defagos) [inline Demo/Sources/Showcase/SourceCodeViewable.swift:32]:
```suggestion
    func sourceCode<T>(of objectType: T.Type) -> some View where T: SourceCodeViewable {
#if os(iOS)
        if let url = gitHubUrl(for: objectType) {
```

File: Demo/Sources/Showcase/SourceCodeViewable.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        else {
+            self
+        }
+#else
+        self
+#endif
+    }
+
+    private func gitHubUrl(for objectType: any SourceCodeViewable.Type) -> URL? {
```
Comment (defagos) [inline Demo/Sources/Showcase/SourceCodeViewable.swift:47]:
```suggestion
    private func gitHubUrl<T>(for objectType: T.Type) -> URL? where T: SourceCodeViewable {
```

**Review by defagos — APPROVED** [review]
Really nice idea, thanks!

File: Demo/Sources/Showcase/SourceCodeViewable.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+import SwiftUI
+
+struct SourceCodeButton: View {
```
Comment (defagos) [inline Demo/Sources/Showcase/SourceCodeViewable.swift:9]:
```suggestion
private struct SourceCodeButton: View {
```

**Review by defagos — APPROVED** [review]

File: Demo/Sources/Showcase/SourceCodeViewable.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+}
+
+protocol SourceCodeViewable {
+    static var filePath: String { get }
+}
+
+extension View {
+    @ViewBuilder
+    func sourceCode<T>(of objectType: T.Type?) -> some View where T: SourceCodeViewable {
```
Comment (defagos) [inline Demo/Sources/Showcase/SourceCodeViewable.swift:27]: Why is an optional introduced here? This fact is not used anywhere in the code and I would always expect to have a type supplied here.

**Review by defagos — APPROVED** [review]

**Pull request merged**